src/script/code-push-error.ts

```diff
@@ -1,13 +1,19 @@
+import {Http} from "./acquisition-sdk";
+
 export class CodePushError extends Error {
-    constructor(message: string) {
-        super(message);
+    constructor(message: string, options?: unknown) {
+        super(
+            message,
+            // @ts-expect-error - need to enable es2022 (see: https://github.com/microsoft/TypeScript/blob/main/src/lib/es2022.error.d.ts)
```

Comment:
What would happen if we use es prior es2022? Should we add a fallback?

Reply:
@DmitriyKirakosyan hi

Reply:
@retyui, let's keep compatibility with older ES.

My suggestion is to add a new member to `CodePushError` class, and then set this member in constructor, rather than `option` parameter in `Error` class. Same thing in `CodePushHttpError` derived class.

This way we encapsulate error in our own class rather than depend on new features of ES2022.